Tasks/Common/webdeployment-common/xmlvariablesubstitutionutility.ts

```diff
@@ -138,60 +141,60 @@ export async function substituteXmlVariables(configFile, tags, variableMap, para
             }  
         }
     }
-    var domContent = (fileEncodeType[1]?'\uFEFF':'') + ltxdomutility.getContentWithHeader(xmlDocument);
-    fs.writeFile(configFile, domContent, fileEncodeType[0], function(error) {
-        if (error) {
-            throw new Error(tl.loc("Failedtowritetoconfigfilewitherror",configFile, error));
-        } else {
-            tl.debug("Config file " + configFile + " updated.");
-        }
-    });
-    
+    var domContent = ( fileEncodeType[1]? '\uFEFF' : '' ) + ltxdomutility.getContentWithHeader(xmlDocument);
+    for(var replacableTokenValue in replacableTokenValues) {
+        tl.debug('Substituting original value in place of temp_name: ' + replacableTokenValue);
```

Comment:
As discussed check on complexity part of algorithm as there is much difference is complexity when substituting in string or in dom.